### ISSUE TITLE
System.Private.CoreLib: Exception while executing function: SetSubscription. Microsoft.Graph.Core: Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by Kike2000** [microsoftgraph/msgraph-training-azurefunction-csharp#13]
**Where did you get the code?**

- [ ] Downloaded from GitHub
- [ ] Downloaded from the [Microsoft Graph quick start tool](https://developer.microsoft.com/graph/quick-start)
- [ ] Followed the tutorial from [Microsoft Graph tutorials](https://docs.microsoft.com/graph/tutorials)

**Describe the bug**
Hi, I'm having problems when making the susbcription with the App Service URL. I don't have any problem when setting NgrokURL when i open an instance of ngrok local 7071 in cmd 

**To Reproduce**
1.-To test i set ngrokUrl value 
![image](https://user-images.githubusercontent.com/72471132/129581476-eaa3e83f-49d5-4349-ae25-b3c5f848df09.png)
2.-Once i set it, i can create a subscription
![image](https://user-images.githubusercontent.com/72471132/129581586-2cc64f93-7793-4a04-be64-4265083c744f.png)
![image](https://user-images.githubusercontent.com/72471132/129582110-81089ad0-d14d-4fec-92de-766874ca9863.png)
3.-But if i replace the ngrok value for 
![image](https://user-images.githubusercontent.com/72471132/129582246-83a1721c-5235-4978-9b76-026e681c7f71.png)
   I get this error
![image](https://user-images.githubusercontent.com/72471132/129582586-fb8ca9e7-ec98-416a-8c1d-5eef1198c566.png)

**Dependency versions**

- Graph library (Graph SDK, REST library, etc.) version 3.8.0

**Comment by jasonjoh** [comment]
Which line of code produces the exception?

**Comment by Kike2000** [comment]
Here Jason, SetSubscription class
![image](https://user-images.githubusercontent.com/72471132/129593143-8aecbbf5-fb78-47c2-8e88-7cc624a66b69.png)

**Comment by jasonjoh** [comment]
Are you running this in a debugger? Can you inspect the values in `subscription`? I'm assuming the `graphClient` is not null?

The value of `ngrokUrl` shouldn't be available to the function running in Azure - it's a user secret that is only available on your dev machine. However, this code should be handling that scenario and setting the notification URL to your function's URL.

https://github.com/microsoftgraph/msgraph-training-azurefunction-csharp/blob/60031af19ee6ab1e5c12d921b7447972bdaeabd6/demo/GraphTutorial/SetSubscription.cs#L50-L54

**Comment by Kike2000** [comment]
So for example if i publish my functions the credentials won't be publish as well? my apiFuncionId and everytime will stay locally?

**Comment by jasonjoh** [comment]
Yes, that's correct. User secrets are for development only. You can read more about this [here](https://docs.microsoft.com/aspnet/core/security/app-secrets). When deployed, you would want to use [App settings](https://docs.microsoft.com/azure/azure-functions/functions-how-to-use-azure-function-app-settings?tabs=portal#settings) for things like app ID and tenant ID, and it's recommended that your secrets be stored in [Azure Key Vault](https://docs.microsoft.com/azure/key-vault/general/about-keys-secrets-certificates).
